Tella/Scenes/GDrive/Views/GdriveReportMainView.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  GdriveReportMainView.swift
+//  Tella
+//
+//  Created by RIMA on 6/9/2024.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import SwiftUI
+struct GdriveReportMainView: View {
+    
+    @ObservedObject var reportsMainViewModel: GDriveViewModel
+    
+    var body: some View {
+        ReportMainView(reportsMainViewModel: reportsMainViewModel) { id in
+            var destination : any View
```

Comment:
could we change this pattern to a method-based navigation with separate functions? I think is more consistent with our codebase and more readable.

it would be something like this:

```
var body: some View {
        ReportMainView(reportsMainViewModel: reportsMainViewModel,
                       showDraftViewAction: navigateToDraftView,
                       showSubmittedViewAction: navigateToSubmittedView,
                       showOutboxViewAction: navigateToOutboxView)
    }
```
and then each method as a separate function:

```
private func navigateToDraftView(_ id: Int?) {
        let draftVM = GDriveDraftViewModel(mainAppModel: reportsMainViewModel.mainAppModel,
                                           repository: viewModel.gDriveRepository,
                                           reportId: id)
        navigateTo(GDriveDraftView(gDriveDraftVM: draftVM, reportsViewModel: viewModel))
    }
```

Reply:
I think is more readable this way but let me know what do you think!

Reply:
Yes, I am currently working on that refactoring locally. I need to do some other changes before